gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -223,7 +225,12 @@ private Object doResumableUpload() throws IOException {
 
       private final long writeOffset;
       private final String uploadId;
+      private final TreeMap<Long, ByteString> dataChunkMap;
       private volatile boolean objectFinalized = false;
```

Comment:
Why do we need volatile? Only a single thread operates in this class at a time, right?

Reply:
Not necessarily, for example the code in beforestart() will be running in a different thread and it could be we want the status of write is reflected in main memory once it's been marked as true in any thread.